cmd/all-in-one/main.go

```diff
@@ -271,11 +275,11 @@ func startQuery(
 	metricsQueryService querysvc.MetricsQueryService,
 	baseFactory metrics.Factory,
 	tm *tenancy.Manager,
-) *queryApp.Server {
+) (*queryApp.Server, jtracer.JTracer) {
 	spanReader = storageMetrics.NewReadMetricsDecorator(spanReader, baseFactory.Namespace(metrics.NSOptions{Name: "query"}))
 	qs := querysvc.NewQueryService(spanReader, depReader, *queryOpts)
-	jtracer := jtracer.OT(opentracing.GlobalTracer())
-	server, err := queryApp.NewServer(svc.Logger, qs, metricsQueryService, qOpts, tm, jtracer)
+	tracer := jtracer.New()
+	server, err := queryApp.NewServer(svc.Logger, qs, metricsQueryService, qOpts, tm, tracer)
```

Comment:
So here server is getting initialised during callback. As we have initiated a otel tracer here, can we replace it initTracer()? If so, how to close this?

Reply:
You create tracer outside of this function and pass it in as parameter.

Reply:
sorry, I didn't get this. I mean I created `tracer` as NoOpTracer

Reply:
You are creating tracer in L103. Just pass it as argument to `startQuery()`, don't create a new one here.